backend/tests/Notifo.Domain.Tests/Channels/MobilePush/UserNotificationExtensionsTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿// ==========================================================================
+//  Notifo.io
+// ==========================================================================
+//  Copyright (c) Sebastian Stehle
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using System.Collections.Generic;
+using FluentAssertions;
+using Notifo.Domain.UserNotifications;
+using Xunit;
+
+namespace Notifo.Domain.Channels.MobilePush
+{
+    public class UserNotificationExtensionsTests
+    {
+        [Fact]
+        public void Should_generate_firebase_message()
+        {
+            string token = "token1";
+            string body = "Lorem ipsum dolor sit amet, consetetur sadipscing elitr";
+            string subject = "subject1";
+            string confirmText = "Got It!";
+            string imageSmall = "https://via.placeholder.com/100";
+            string imageLarge = "https://via.placeholder.com/600";
+            string trackingUrl = "https://track.notifo.com";
+            string confirmUrl = "https://confirm.notifo.com";
+
+            var notification = new UserNotification
+            {
+                Formatting = new NotificationFormatting<string>
+                {
+                    Body = body,
+                    Subject = subject,
+                    ImageSmall = imageSmall,
+                    ImageLarge = imageLarge,
+                    ConfirmText = confirmText,
+                    ConfirmMode = ConfirmMode.Explicit
+                },
+                TrackingUrl = trackingUrl,
+                ConfirmUrl = confirmUrl
+            };
+
+            var message = notification.ToFirebaseMessage(token);
+
+            message.Token.Should().BeEquivalentTo(token);
```

Comment:
If you use fluent assertions (which I only use in exceptions) it is better to compare the whole object. Because fluent assertions can make deep comparisons.